### PR TITLE
Serving LA screen changes SOA

### PR DESCRIPTION
**Pull request by BharadwajSairam**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PRL-4159

### Change description ###

Serving LA screen changes SOA

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java [hunk 1]
```diff
@@ -843,9 +847,19 @@ public Map<String, Object> getSoaCaseFieldsMap(CaseDetails caseDetails) {
...
+        caseDataUpdated.put("soaDocumentDynamicListForLa", getDocumentsDynamicListForLa(authorisation,
+                                                                                        String.valueOf(caseData.getId())));
+        log.info("**SOA options *** {}",caseDataUpdated);
         return caseDataUpdated;
     }
 
+    private List<Element<DocumentListForLa>> getDocumentsDynamicListForLa(String authorisation, String caseId) {
+        return List.of(Element.<DocumentListForLa>builder().id(UUID.randomUUID()).value(DocumentListForLa.builder()
+                                                                                      .documentsListForLa(sendAndReplyService
+                                                                .getCategoriesAndDocuments(authorisation, caseId))
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java:859]: can the following be moved to utility? getCategoriesAndDocuments()

**Review by shashisk97 — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -571,7 +571,8 @@ def sonarExclusions = [
   '**/uk/gov/hmcts/reform/prl/services/TestingSupportService.java',
   '**/uk/gov/hmcts/reform/prl/mapper/bundle/BundleCreateRequestMapper.java',
   '**/uk/gov/hmcts/reform/prl/controllers/DraftAnOrderController.java',
-  '**/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java'
+  '**/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java',
+  '**/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java'
```
Comment (shashisk97) [inline build.gradle:575]: remove exclusion?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationEmailService.java [hunk 3]
```diff
@@ -238,4 +240,18 @@ private EmailTemplateVars buildApplicantEmailVars(CaseData caseData, String appl
             .build();
 
     }
+
+    public EmailNotificationDetails sendEmailNotificationToLocalAuthority(String authorization, CaseData caseData,
+                                                                          String email,
+                                                                          List<Document> docs,String servedParty) throws IOException {
+        return sendgridService.sendEmailWithAttachments(authorization,
+                                                        getEmailProps(
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationEmailService.java:248]: Change this to send only required fields in email props

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java [hunk 4]
```diff
@@ -339,6 +352,21 @@ public ServedApplicationDetails sendNotificationForServiceOfApplication(CaseData
...
+        log.info("Sending notifiction to LA");
+        if (docsForLa.size() != 0) {
+            try {
+                emailNotificationDetails.add(element(serviceOfApplicationEmailService
+                                                         .sendEmailNotificationToLocalAuthority(authorization,
+                                                                                                caseData,
+                                                                                                caseData.getServiceOfApplication()
+                                                                                                    .getSoaLaEmailAddress(),
+                                                                                                docsForLa,
+                                                                                                PrlAppsConstants.SERVED_PARTY_CAFCASS_CYMRU)));
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java:365]: is it cafcass cymru?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java [hunk 5]
```diff
@@ -849,6 +966,9 @@ public Map<String, Object> getSoaCaseFieldsMap(CaseDetails caseDetails) {
             || CaseUtils.isC8Present(caseData) ? Yes : No);
         caseDataUpdated.put(CASE_TYPE_OF_APPLICATION, CaseUtils.getCaseTypeOfApplication(caseData));
         caseDataUpdated.put(CASE_CREATED_BY, caseData.getCaseCreatedBy());
+        caseDataUpdated.put("soaDocumentDynamicListForLa", getDocumentsDynamicListForLa(authorisation,
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java:969]: move to constant

**Review by yogendraU — APPROVED** [review]

**Pull request merged**